src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -1010,8 +1011,9 @@ def create_python_udf_or_sp(
         if inline_python_code
         else ""
     )
-
+    mutability = "IMMUTABLE" if immutable else "VOLATILE"
```

Comment:
depending on `immutable` being optional, I think this would be

```python
mutability = ""
if immutable is not None:
    mutability = "IMMUTABLE" if immutable else "VOLATILE"

Reply:
If we agree on the default VOLATILE behavior, then we don't need this :)